site-src/concepts/guidelines.md

```diff
@@ -38,6 +38,41 @@ prescribed resolution, the following guiding principles should be applied:
   should be given to the resource appearing first in alphabetical order
   (namespace/name). For example, foo/bar would be given precedence over foo/baz.
 
+## Gracefully Handling Future API Versions
+
+An important consideration when implementing this API is how it might change in
+the future. Similar to the Ingress API before it, this API is designed to be
+implemented by a variety of different products within the same cluster. That
+means that the API version your implementation was developed with may be
+different than the API version it is used with. At a minimum, the following
+requirements must be met to ensure future versions of the API do not break your
+implementation:
+
+* Handle fields with loosened validation without crashing
+* Handle fields that have transitioned from required to optional without
+  crashing
+
+## Limitations of CRD and Webhook Validation
+
+CRD and webhook validation is not the final validation i.e. webhook is "nice UX"
+but not schema enforcement. This validation is intended to provide immediate
+feedback to users when they provide an invalid configuration. Write code
+defensively with the assumption that at least some invalid input (Gateway API
+resources) will reach your controller. Both Webhook and CRD validation is not
+fully reliable because it:
```

Comment:
(note: this is a generic CRD/webhook problem, not just for this specific API)

Reply:
(would be good to make sure this is clear)

Reply:
Updated, thanks!